src/util-hugepages.c

```diff
@@ -0,0 +1,330 @@
+/* Copyright (C) 2023 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Lukas Sismis <lsismis@oisf.net>
+ */
+
+#include "suricata.h"
+#include "util-debug.h"
+#include "util-hugepages.h"
+
+#include <dirent.h>
+
+#if !defined __CYGWIN__ && !defined OS_WIN32 && !defined __OpenBSD__ && !defined sun
```

Comment:
can you explain why we have these specific guards here?

Reply:
so that it is only available on Linux. 
I haven't found a better way in the codebase so I can rework it if it can be done better.

Reply:
I don't necessarily think we need guards at all, unless we get compiler failures due to linux specific calls. But if we can keep things limited to runtime error handling, that would be enough.

Reply:
We have compiler failures, see the changelog in https://github.com/OISF/suricata/pull/9898

Reply:
But I refactored the code and removed the guards on as many places as I could